44 guest login google creds master

akintner opened the pull request:
- all functionality present for guests to login and/or register with either google oauth credentials or pinspiration credentials

akintner committed:
initial setup for google oauth, add routes, update layout meta tags

akintner committed:
continue to try google_oauth

mollybrown committed:
add google oauth2 authorization functionality. Callback works, next step is to coerece the correct params out of the response and save them as a registered user's google credential.

akintner committed:
update google oauth functionality

akintner committed:
update final google_credentials functionality

akintner committed:
finalize tests for user registration with both google and pinspiration credentials, update google_credential model test to account for the fact that phone_numbers will never be coming through from google

akintner committed:
Merge branch 'master' into 44_guest_login_google_creds_master

akintner commented:
We will eventually have to re-do some of the tests here once we have different redirect paths for where a user is actually sent once they login (this might be ready already, but I'm not entirely sure) because at the moment we're just redirecting to the root_path.

wlffann commented on app/models/google_credential.rb:
```diff
@@ -2,10 +2,24 @@ class GoogleCredential < ApplicationRecord
+        credential = registered_user.google_credentials.create(google_uid: auth_info.info.google_uid, 
+                                                  name: auth_info.info.name, 
+                                                  username: auth_info.info.name.parameterize)
+        credential
```
Have investigated pulling emails from google fully? I think we definitely need them since our username method will likely use them

akintner replied:
Email definitely isn't coming back in the OAuth hash. If there's a way to extract them from the actual login action, I'm not aware of it, though I agree that we'll need it and if anyone has ideas of wants to take a stab at it, feel free.

wlffann commented on config/secrets.yml:
```diff
@@ -12,10 +12,13 @@
   secret_key_base: 0f795c1ada24ff4780d43bb2ad7896eee86616728ccb7e5b0f5b924149991e73ac51140e3e35b69bfd4df106bda484a797c2ba3c7501e24cb29b0e159ccef4e7
-
+  google_client_id: 757656209931-n8t7v63pfl907i02rt96brrfi4nil4ak
+  google_client_secret: AYt1TuBizF1MF1ZJHDxAKmjq
```
Should these be pushed?

akintner replied:
Definitely no, though I have both client_id and secret in application.yml, which is gitignored. Not sure where this came from.

akintner replied:
Will investigate.

akintner replied:
@mollybrown any ideas? In the meantime, I'll gitignore this and re-push this branch.

akintner replied:
So I just updated the gitignore and saw this when I typed `git status` in the terminal: 
Last commands done (5 commands done):
   pick c2db04a Add admin logout
   pick 8506e8a Add model validations, associations, and tests
  (see more in file .git/rebase-merge/done)
Next command to do (1 remaining command):
   pick 507b91a Add admin login
  (use "git rebase --edit-todo" to view and edit)
You are currently editing a commit while rebasing branch 'styling_sass' on '553992f'.
  (use "git commit --amend" to amend the current commit)
  (use "git rebase --continue" once you are satisfied with your changes)

Is there an active rebase in progress for this branch?

CPowell23 replied:
Oh dear :) Just now seeing this. Good catch Annie

CPowell23 replied:
@akintner - are we able to deactivate and reset the keys?

akintner replied:
They should be all good now, yes

wlffann reviewed (commented):
I think there are some issues to resolve before we merge this. Maybe the tests should even be added to this PR?

akintner committed:
add gem figaro and ammend gitignore

akintner committed:
rebase with master to update
Merge branch '44_guest_login_google_creds_master' of https://github.com/CPowell23/pinspiration into 44_guest_login_google_creds_master

akintner committed:
Merge branch 'master' into 44_guest_login_google_creds_master

CPowell23 commented on app/controllers/sessions_controller.rb:
```diff
@@ -10,16 +10,15 @@ def create
+  end
+
+  def destroy
+    session[:user_id] = nil
```
I think this should be session[:registered_user_id} = nil

akintner commented:
I'm closing this because of the issues with the keys being here. Going to reopen it in a sec and make sure nothing fancy/secret is being pushed.
